pandas/tests/frame/test_missing.py

```diff
@@ -425,3 +425,24 @@ def test_fill_corner(self):
 
         # TODO(wesm): unused?
         result = empty_float.fillna(value=0)  # noqa
+
+    def test_fill_value_missing_with_binary_operators(self):
+        # GH12723
+        dat = np.arange(10)
```

Comment:
specify this as ``np.arange(10, dtype='int64')`` or the comparison will fail on windows